gateway/tests/e2e/providers/openai.rs

```diff
@@ -160,3 +161,24 @@ async fn test_o1_mini_inference() {
     let raw_response = result.get("raw_response").unwrap().as_str().unwrap();
     let _raw_response_json: Value = serde_json::from_str(raw_response).unwrap();
 }
+
+#[tokio::test]
+async fn test_embedding_request() {
+    let provider_config_serialized = r#"
+    type = "openai"
+    model_name = "text-embedding-3-small"
+    "#;
+    let provider_config: EmbeddingProviderConfig = toml::from_str(provider_config_serialized)
+        .expect("Failed to deserialize EmbeddingProviderConfig");
+    assert!(matches!(
+        provider_config,
+        EmbeddingProviderConfig::OpenAI(_)
+    ));
+
+    let client = Client::new();
+    let request = EmbeddingRequest {
+        input: "This is a test input".to_string(),
+    };
+    let response = provider_config.embed(&request, &client).await.unwrap();
+    assert_eq!(response.embedding.len(), 1536);
```

Comment:
1. Check every field
2. Check embedding is not zero vector or some other trivial case.
3. Add a sanity check: test two similar words `a, b` and a random word `c` (all predefined). `d(a, b) << d(a, c)`